into/backends/aws.py

```diff
@@ -0,0 +1,184 @@
+    >>> S3(CSV)
+    S3(CSV)
+    >>> S3(sqlalchemy.Table)
+    """
+    return type('S3(%s)' % cls.__name__, (cls, _S3), {'subtype': cls})
+
+
+@resource.register('s3://.*\.csv', priority=18)
+def resource_s3_csv(uri):
+    return S3(CSV)(uri)
+
+
+@convert.register(CSV, S3(CSV))
+def convert_s3_to_csv(csv, **kwargs):
+    """Download an S3 CSV to a local CSV file."""
+    path = csv.path
+    dirname, basename = os.path.split(path)
+    dirname = remove_substring(dirname)
+
+    mkdir_p(dirname)
+
+    conn_kwargs = dict(aws_access_key_id=kwargs.pop('aws_access_key_id', None),
+                       aws_secret_access_key=(
+                                    kwargs.pop('aws_secret_access_key', None)))
+    conn = get_s3_connection(**conn_kwargs)
+    bucket = conn.get_bucket(dirname)
+    key = bucket.get_key(basename)
+    filename = os.path.join(dirname, basename)
+    key.get_contents_to_filename(filename)
+    return CSV(filename, **kwargs)
+
+
+def get_part(bucket, key, conn=None, start=0, stop=8192,
+             aws_access_key_id=None, aws_secret_access_key=None, **kwargs):
+    """Get bytes from `start` to `stop` of `key` from the S3 bucket `bucket`.
+
+    Parameters
+    ----------
+    bucket : str
+        Bucket name
+    key : str
+        File name
+    conn : Connection, optional, default ``None``
+        S3 connection
+    start : int, optional, default ``0``
+        Byte to start at, zero based
+    stop : int, optional, default ``8192``
+        Byte to stop at, zero based
+    aws_access_key_id : str, optional, default ``None``
+        Access key id. Will be taken from the environment if ``None``.
+    aws_secret_access_key : str, optional, default ``None``
+        Secret access key. Will be taken from the environment if ``None``.
+    kwargs : dict
+        Keyword arguments to pass to the CSV constructor
+
+    Returns
+    -------
+    csv : CSV
+        The ``stop - start`` bytes of the S3 bucket named
+        ``s3://<bucket>/<key>`` as a CSV file.
+
+    Examples
+    --------
+    >>> csv = get_part(bucket='nyqpug', key='tips.csv')
+    >>> print(discover(csv))
+    var * {
+      total_bill: float64,
+      tip: float64,
+      sex: string,
+      smoker: string,
+      day: string,
+      time: string,
+      size: int64
+      }
+    """
+    assert stop - start > 0
+
+    if conn is None:
+        conn = get_s3_connection(aws_access_key_id=aws_access_key_id,
+                                 aws_secret_access_key=aws_secret_access_key)
+
+    # TODO: is Range always accepted by S3?
+    # TODO: is Content-Type useful here?
+    resp = conn.make_request('GET',
+                             bucket=bucket,
+                             key=key,
+                             headers={'Range': 'bytes=%d-%d' % (start, stop)})
+    raw = resp.read()
+    raw = raw[:raw.rindex(b'\n')]
+    with tempfile.NamedTemporaryFile(mode='wb', delete=False) as f:
+        f.write(raw)
+    return CSV(f.name, **kwargs)
```

Comment:
Should maybe be a `contextmanager` to remove the temporary file?  My HDFS solution to this is here https://github.com/ContinuumIO/into/pull/44/files#diff-336aefd0b6a5fb97d9a47767a30175d8R56